### DISCUSSION TITLE
Issue using SBFspot in parallel with Domoticz

### DISCUSSION THREAD

**Discussion by pa3fat** [SBFspot/SBFspot#609] [Q&A]
Since number of days i have an issue with SBFspot 3.3.1 running in parallel with Domoticz.
Unclear at this moment what is the cause but running SBFspot -v i found this below errors.
Restarting the Raspberry pi everything starts working again but it stops due to yet unknown reasons.

If possible (and if this is the right support area, let me know if there is a forum) I would like to know what could be the cause.
How to resolve any issues. Eg check if there is an sd card issue. Any help is appreciated. 
Domoticz forum states there were no changes in the SBFSpot usage within Domoticz.

Commant to check:
/home/pi/SBFspot/SBFspot/bin/Release_SQLite/SBFspot -v
Current Inverter Time: 06/01/2023 13:59:40
Inverter Wake-Up Time: 06/01/2023 09:13:50
Inverter Sleep Time : 06/01/2023 13:59:43
ExportSpotDataToCSV()
[2023-01-06 13:59:43.999] Error: sqlite3_exec() returned: 'disk I/O error' while executing
INSERT OR IGNORE INTO Inverters VALUES(2130112788,'SN: 2130112788','SB 5000TL-21','02.60.04.R',0,0,0,0,0,0,'','',0)
[2023-01-06 13:59:44.000] Error: exec_query() returned: 'disk I/O error' while executing
INSERT OR IGNORE INTO Inverters VALUES(2130112788,'SN: 2130112788','SB 5000TL-21','02.60.04.R',0,0,0,0,0,0,'','',0)
[2023-01-06 13:59:44.001] Error: sqlite3_exec() returned: 'disk I/O error' while executing
UPDATE Inverters SET Name='SN: 2130112788',Type='SB 5000TL-21',SW_Version='02.60.04.R' WHERE Serial=2130112788
[2023-01-06 13:59:44.001] Error: exec_query() returned: 'disk I/O error' while executing
UPDATE Inverters SET Name='SN: 2130112788',Type='SB 5000TL-21',SW_Version='02.60.04.R' WHERE Serial=2130112788
[2023-01-06 13:59:44.002] Error: sqlite3_exec() returned: 'disk I/O error' while executing
UPDATE Inverters SET TimeStamp=1673009983,TotalPac=490,EToday=1.639,ETotal=37014.8,OperatingTime=38133.1,FeedInTime=36986.4,Status='OK',GridRelay='Closed',Temperature=30.94 WHERE Serial=2130112788
[2023-01-06 13:59:44.003] Error: exec_query() returned: 'disk I/O error' while executing
UPDATE Inverters SET TimeStamp=1673009983,TotalPac=490,EToday=1.639,ETotal=37014.8,OperatingTime=38133.1,FeedInTime=36986.4,Status='OK',GridRelay='Closed',Temperature=30.94 WHERE Serial=2130112788
[2023-01-06 13:59:44.004] Error: sqlite3_exec() returned: 'disk I/O error' while executing
INSERT INTO SpotData VALUES(1673009983,2130112788,258,250,0.824,0.791,313.34,317.47,489,0,0,2.118,0,0,231.69,0,0,1639,37014804,50.01,38133.1,36986.4,67.451,'OK','Closed',30.94)
[2023-01-06 13:59:44.005] Error: [spot_data]exec_query() returned: 'disk I/O error' while executing
INSERT INTO SpotData VALUES(1673009983,2130112788,258,250,0.824,0.791,313.34,317.47,489,0,0,2.118,0,0,231.69,0,0,1639,37014804,50.01,38133.1,36986.4,67.451,'OK','Closed',30.94)
********************
* ArchiveDayData() *
********************
startTime = 63B75670 -> 06/01/2023 00:00:00
ExportDayDataToCSV()
**********************
* ArchiveMonthData() *
**********************
startTime = 63B167B0 -> 01/01/2023 12:00:00
ExportMonthDataToCSV()
Reading events: 2023-Jan-01
ExportEventsToCSV()
Fri Jan 6 13:59:44 2023: INFO: Done.

**Comment by sbf-** [comment]
Most probably a corrupt SD-card

**Reply by pa3fat** [reply]
Any option to 'repair'?

**Reply by pa3fat** [reply]
Domoticz is still running ok. 
Could i just replace the 'damaged' SBFspot.db? And how?

**Reply by pa3fat** [reply]
if it is empty is ok. But long time ago when i installed this. And not really into this os...
Is there a specific command to get me a new empty db?

**Reply by pa3fat** [reply]
did some checks via sqlite3 and reports io errors.
made a copy of the exisiting db and generated a new one following the wiki.
sqlite3 SBFspot.db < ~/SBFspot/SBFspot/CreateSQLiteDB.sql
checked this db with PRAGMA integrety_check. IO errors are gone.

Rebooting the PI to see if this resolved it.
Also noticed that there was a space error before creating the db. Removed the 2020 directory. That helped. Will do some more clearning of this old stuff.

**Reply by sbf-** [reply]
Good job, but to me it doesn't look good future wise.
Once the SD gives errors, it's almost dead. So, be prepared.

**Comment by pa3fat** [comment]
Had a planned but not so well done power outage around 23rd of Dec. Probably there the issue started.
Was trying to identify radio interference on 3.7Mhz. 

Will read in how to make a full backup of the sd to a new sd via an usb adapter. 

Still running ok.
